x-pack/plugins/apm/server/routes/service_map/get_trace_sample_ids.ts

```diff
@@ -60,7 +63,7 @@ export async function getTraceSampleIds({
   let events = [ProcessorEvent.span, ProcessorEvent.transaction];
 
   // perf optimization that is only possible on the global service map with no filters
-  if (isGlobalServiceMap) {
+  if (isGlobalServiceMap && !(kuery || serviceGroupKuery)) {
```

Comment:
Looks like this is repeating part of what's already in `isGlobalServiceMap`

```suggestion
  if (isGlobalServiceMap) {
```

What about updating `isGlobalServiceMap`:

```ts
const isGlobalServiceMap = !serviceName && !serviceGroupKuery && !kuery;
```

Perhaps rename it to `isUnfilteredGlobalServiceMap`